pkgs/by-name/li/likwid/package.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, stdenv
+, fetchurl
+, perl
+, coreutils
+, gnugrep
+}:
+
+stdenv.mkDerivation rec {
+  pname = "likwid";
+  version = "5.3.0";
+
+  src = fetchurl {
+    url = "https://ftp.fau.de/pub/likwid/likwid-${version}.tar.gz";
+    hash = "sha256-wpDlVMQlMSSsKriwVuFO5NI5ZrjJ+/oQuoH3WuVDzk4=";
+  };
+
+  nativeBuildInputs = [ perl ];
+
+  hardeningDisable = [ "format" ];
+
+  preBuild = ''
```

Comment:
1. Put this in `postPatch`
2. You should write a patch file instead of rewriting C code
3. Instead of heavy patching, it is easier to put coreutils and gnugrep as buildInputs.

Reply:
Point 1. is done.
Regarding point 2., I’ve used patch files & `substituteAll` instead `substituteInPlace`.
For point 3., see my earlier post below.